src/domain/datasets/services/src/dataset_key_value_service_sys_env.rs

```diff
@@ -35,15 +36,28 @@ impl DatasetKeyValueServiceSysEnv {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#[async_trait::async_trait]
 impl DatasetKeyValueService for DatasetKeyValueServiceSysEnv {
-    async fn find_dataset_env_var_value_by_key<'a>(
+    fn find_dataset_env_var_value_by_key(
         &self,
         dataset_env_var_key: &str,
-        _dataset_env_vars: &'a HashMap<String, DatasetEnvVar>,
+        dataset_env_vars: &HashMap<String, DatasetEnvVar>,
```

Comment:
I found the code of this service extremely verbose. If we are in `DatasetKeyValueService` and in method called `find_dataset_env_var_value_by_key` instead of `dataset_env_var_key` we could call parameter `key` without any ambiguity.